types/index.d.ts

```diff
@@ -0,0 +1,54 @@
+// TypeScript Version: 3.0
+
+import {Node, Parent} from 'unist'
+import {Test} from 'unist-util-is'
+
+declare namespace mdastUtilToc {
+  interface Link extends Parent {
+    type: 'link'
+    title: null
+    url: string
+    children: Node[]
+  }
+
+  interface Paragraph extends Parent {
+    type: 'paragraph'
+    children: Link[]
+  }
+
+  interface ListItem extends Parent {
+    type: 'listItem'
+    loose: boolean
+    spread: boolean
+    children: Node[]
+  }
+
+  interface List extends Parent {
+    type: 'list'
+    ordered: boolean
+    spread: boolean
+    children: ListItem[]
+  }
+
+  interface TOCOptions {
+    heading?: string
+    maxDepth?: number
+    skip?: string
+    tight?: boolean
+    prefix?: string
+    parents?: Test<Node> | Array<Test<any>>
+  }
+
+  interface TOCResult {
+    index?: number
+    endIndex?: number
+    map?: List
+  }
+}
+
+declare function mdastUtilToc(
```

Comment:
TSDocs can be helpful for adopters, the docs show next to the types for any editor that uses the [typescript language server (mainly VSCode)](https://github.com/microsoft/vscode/tree/master/extensions/typescript-language-features) 

```suggestion
/**
 * Generate a Table of Contents from a tree.
 *
 * @param node searched for headings
 * @param options configuration and settings
 */
declare function mdastUtilToc(
```